### DISCUSSION TITLE
await code fix / analyzer enhancement

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/roslyn#60285] [Ideas]
Today, if you have a method like this:

```
public Task FooAsync() {

  return Task.Completed;
}
```

and you type "await " in the body of the method, the signature of the method automatically changes to include "async" so that it reads as so:

```
public async Task FooAsync() { // Notice that async was added.

  await //this is where I was typing

  return Task.CompletedTask;
}
```

This is great but it would be even better if a few other changes would happen:
1.  Change any "return Task.CompletedTask" to just "Return" (or, perhaps, omit it completely depending on location)
2. Change any "Task.FromResult(x)" to just x

**Comment by CyrusNajmabadi** [comment]
I like these ideas.  @sharwell thoughts?

**Reply by Youssef1313** [reply]
Yeah I like that too. @CyrusNajmabadi Could this be converted to an issue for better tracking?

**Reply by CyrusNajmabadi** [reply]
Done: https://github.com/dotnet/roslyn/issues/60318
